src/main/scala/Exercise8_4.scala

```diff
@@ -0,0 +1,36 @@
+object Exercise8_4 extends App {
+  abstract class Item {
+    def price:Double
+    def description:String
+  }
+  class SimpleItem(val price:Double, val description:String) extends Item
+
+  class Bundle(val bundlePrice:Double, val bundleName:String) extends Item {  // An item able to store simple items
+
+    private var content:Array[SimpleItem] = new Array(0) // Stored items. I prefer not to stack bundles
+
+    def price:Double = bundlePrice + content.map(_.price).sum  // price of the bundle itself + content's total price
+
+    def description:String = { // description of bundle and its content
+      if (content.length == 0)
+        "An empty " + bundleName
+      else
+        bundleName + " with one " + content.map(_.description).mkString(", one ") + " inside."
+    }
+
+    def AddSimpleItem(newConts:SimpleItem*) { // a method to add items to the bundle
+      for (newCont <- newConts) content = content :+ newCont
```

Comment:
Since you use `:+`, I think you can just use `++`. Try it out.
```suggestion
      content ++ newConts
```

Reply:
I can write:
22        content = content ++ newConts
If not, content is not overwritten.

Reply:
You're right, good point.

If we had had unit tests, by changing this, we had added a bug which we would have detected it. 👍